PyPDF2/_merger.py

```diff
@@ -252,10 +264,6 @@ def write(self, fileobj: StrByteType) -> None:
         """
         if self.output is None:
             raise RuntimeError(ERR_CLOSED_WRITER)
-        my_file = False
-        if isinstance(fileobj, str):
-            fileobj = FileIO(fileobj, "wb")
-            my_file = True
```

Comment:
Removing this (and lines below) will break existing usage. We want to support both the contextmanager and existing workflows.